### PR TITLE
Fixed gallery button insertion script and design fix

### PR DESCRIPTION
**Pull request by skjnldsv**
Fixes nextcloud/server#7374
Needs nextcloud/server#7478

dev skjnldsv com_40402_apps_files__di156r _
dev skjnldsv com_40402_apps_files__dir _

### CODE REVIEW

**Comment by MorrisJobke** [comment]
The gallery controls look still a bit weird:

bildschirmfoto 2017-12-13 um 12 16 48

**Comment by skjnldsv** [comment]
Ouuuups, let me fix it

**Comment by oparoz** [comment]
When testing, it's important to also try with public shares as the layout is completely different.

**Comment by pixelipo** [comment]
3 issues visible here:
image

- [ ] folder breadcrumb is not in line with the rest
- [ ] some icons not in line with the rest
- [ ] icons in `right` container are not pushed to the right edge (flex is needed here as well plus something to account for no sidebar...)
- [ ] It's even worse (but not visible here) once the loader is on:
screenshot from 2017-12-13 13-15-30

In general, button in that controls bar seem to each have their own CSS ruleset. It would be nice to get them to work with as little custom css as possible

**Comment by skjnldsv** [comment]
dev skjnldsv com_40402_apps_gallery_
Fixed a lot, please re-test @pixelipo :)

File: css/styles.css [hunk 1]
```diff
@@ -29,8 +29,12 @@ div.crumb.last a {
 	width: 40px;
 }
 
-#controls > .right {
-	float: right;
+#controls .button:not(:last-child) {
```
Comment (pixelipo) [inline css/styles.css]: this should go to https://github.com/nextcloud/server/pull/7478

Comment (skjnldsv) [reply css/styles.css]: Gallery uses a lot of similar styles but does not includes any of them (yes, we need to rebuild the entire app), therefor, there are a lot of multiples files/properties used.

**Comment by skjnldsv** [comment]
>  #controls is 44px high, so margin-top for buttons should be 4px (or 2px if we go for 40x40 square buttons)

Buttons are always 44x44 on nextcloud! :)
I'll fix it asap

**Comment by pixelipo** [comment]
@skjnldsv be careful with 44x44 - it means we also need to make `#controls {height: 50px}` and `#controls button {margin-top: 3px)` - so that's a commit for https://github.com/nextcloud/server/issues/7374

**Comment by skjnldsv** [comment]
Well, I guess it won't be 44x44 then! I don't want to try and break everything! :O 
😭

**Comment by pixelipo** [comment]
@skjnldsv ok, I've created a new issue for that: https://github.com/nextcloud/server/issues/7481

File: css/styles.css (hunk 1 above)
Comment (pixelipo) [reply css/styles.css]: :cry:

**Comment by skjnldsv** [comment]
>  breadscrumbs folders are still not vertically aligned

What do you mean by that?

**Comment by pixelipo** [comment]
@skjnldsv look here - folder breadcrumb is 2px lower than the rest:
screenshot from 2017-12-13 15-19-57

**Comment by skjnldsv** [comment]
@pixelipo And it's the opposite on the files app, the buttons are 2px lower...

**Comment by pixelipo** [comment]
@skjnldsv that problem would be solved with proper button size and margins

**Comment by skjnldsv** [comment]
@pixelipo the gallery app doesn't even use the new breadcrumbs system, which could be why it's bugged. I'm a bit lost here (and today is my weekend,so I will see this later 😝 ) If you want to investigate a little bit further, please do ;)

**Comment by oparoz** [comment]
> @pixelipo the gallery app doesn't even use the new breadcrumbs system

Very true, by design apps should never rely on components delivered in the server, which is annoying.

**Comment by MorrisJobke** [comment]
I pushed a fix to this branch to properly work with nextcloud/server#7484  this is now good to be merged. 👍

**Comment by MorrisJobke** [comment]
> breadscrumbs folders are still not vertically aligned

Let's skip this for this PR.

**Comment by MorrisJobke** [comment]
I didn't had the second commit from @skjnldsv (27f4c4e9804e608ad56d31b1f2856a8351ed3072) - with this the file switcher is now misplaced. Without that commit it is properly shown.

**Comment by pixelipo** [comment]
@MorrisJobke I'll make another PR in a second

File: css/styles.css [hunk 2]
```diff
@@ -20,16 +17,19 @@ body:after {
 	transition: transform .1s ease-in;
 }
 
-div.crumb.last a {
-	cursor: default;
+/* Two ugly hacks to fix breadcrumbs in Gallery */
```
Comment (skjnldsv) [inline css/styles.css]: shouldn't this go in the gallery then? :)

**Comment by pixelipo** [comment]
That was last commit from be - it fixes public view. Before:
image

After:
image

**Comment by skjnldsv** [comment]
@pixelipo you're awesome! Perfect here!

**Comment by pixelipo** [comment]
Sorry, I lied - I found out that my "ugly hack" was causing a bug when only "Home" is the only breadcrumb. I've fixed it now, with the downside (which cannot be fixed without total rewrite of Gallery breadcrumbs) is that in case ellipsis is the last breadcrumb, it doesn't have the expected margin-right.

Good:
image

Still good:
image

Bad:
image

I guess we can live with that for now ;)

**Comment by pixelipo** [comment]
My last (no, really) commit fixes the hover inconsistency issue mentioned by @MariusBluem in the #7374

**Comment by skjnldsv** [comment]
Another bump here?

**Comment by pixelipo** [comment]
@skjnldsv travis-ci is failing:
```
1) ConnectWithTokenCest: Make sure i can get the thumbails
 Test  tests/api/ConnectWithTokenCest.php:connectToThumbnails
 Step  See response contains ""fileid":"274","status":200"
 Fail  REST response contains
Failed asserting that 'event: preview
data: {"preview":null,"mimetype":"application\/postscript","fileid":"274","status":500}
```

**Comment by oparoz** [comment]
> @skjnldsv travis-ci is failing:

Unrelated. I still haven't found the time to trace the regression. Could have been broken by a change in the server. There is an issue opened about it.

**Comment by skjnldsv** [comment]
@MorrisJobke @rullzer A review here? 🙏

**Comment by skjnldsv** [comment]
13+ I think. We need to test on stable12.

**Comment by danxuliu** [comment]
>  Does the gallery app now only support 13+ or is this still working with 12 and before?

Even if this pull request worked on 12 (and I made a quick check and there seems to be some issues) due to #331 the slideshow icons are not visible on a black background on Nextcloud 12:
gallery-slideshow-buttons

**Comment by rullzer** [comment]
Ok lets bump to 13 only then we have the stable branches anyways

**Pull request merged**